libbeat/template/template.go

```diff
@@ -17,14 +17,15 @@ var (
 )
 
 type Template struct {
-	index       string
-	beatVersion Version
-	esVersion   Version
-	settings    common.MapStr
+	index          string
+	beatVersion    Version
+	esVersion      Version
+	settings       common.MapStr
```

Comment:
I can imagine there could be more yes (e.g. _all), although I don't plan more at the moment. We could maybe use `index` for `x`, so you'd have: `settings.index.number_of_shards` and `settings._source.enabled: false`. Trying to figure out if there can be any settings that do not fall under `index`...